### PR TITLE
Gatekeeper patch.

### PR DESCRIPTION
**Pull request by ndev2**
Pull request for the Gatekeeper patch in tc(8) userspace.

### CODE REVIEW

**Comment by ndev2** [comment]
Hi Michel, the userspace patch is ready from my end, let me know if need to make any changes. Hope the code is good, thanks!

File: man/man8/tc-skbprio.8 [hunk 1]
```diff
@@ -0,0 +1,85 @@
...
+.B tc qdisc ... add skbprio
+.B [ limit
+packets
+.B ]
+
+.SH DESCRIPTION
+The SKB Priority Queue is a queueing discipline, intended to be used as an
+open source
+.B DoS
+defense software. It sends traffic based on an assigned priority for the
```
Comment (AltraMayor) [inline man/man8/tc-skbprio.8]: Improving the first sentence from:
`The SKB Priority Queue is a queueing discipline, intended to be used as an open source DoS defense software.`
to
`The SKB Priority Queue is a queueing discipline intended to prioritize packets during a DoS attack.`

Comment (AltraMayor) [reply man/man8/tc-skbprio.8]: Improving the second sentence from:
`It sends traffic based...`
to:
`It prioritizes traffic based...`

File: tc/q_skbprio.c [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+{
+	struct tc_skbprio_qopt *qopt;
+
+	if (opt == NULL)
+		return 0;
+
+	if (RTA_PAYLOAD(opt)  < sizeof(*qopt))
+		return -1;
+	qopt = RTA_DATA(opt);
+	fprintf(f, "limit %up ", qopt->limit);
```
Comment (AltraMayor) [inline tc/q_skbprio.c]: Why do we need the `p` in the format?

File: man/man8/tc-skbprio.8 (hunk 1 above)
Comment (AltraMayor) [reply man/man8/tc-skbprio.8]: Could give me push access to this branch so GitHub allows me to edit this text?

**Review by AltraMayor — CHANGES_REQUESTED** [review]
**skbprio patch**

File: man/man8/tc-skbprio.8 (hunk 1 above)
Comment (ndev2) [reply man/man8/tc-skbprio.8]: Hi Michel, here's the invite link: https://github.com/ndev2/iproute2-final/invitations

File: tc/q_skbprio.c (hunk 2 above)
Comment (ndev2) [reply tc/q_skbprio.c]: The `p` is to symbolize that it is in packets (as opposed to bytes). Should we change this?

Comment (AltraMayor) [reply tc/q_skbprio.c]: As you can read in Cong's comment to Qiaobin's patch (https://patchwork.ozlabs.org/patch/941073/): "Please align the output syntax with the input syntax. Traditionally iproute2 command output can be just used for input, even though
it is no longer true now."

So the `p` above is breaking this alignment.

File: man/man8/tc-skbprio.8 (hunk 1 above)
Comment (AltraMayor) [reply man/man8/tc-skbprio.8]: I've edited the text. Please merge my patches into yours and review the text as well as the format of the man page.

**Comment by ndev2** [comment]
Hi @AltraMayor , I've pulled and rebased with somw changes, hope the code is good. Thanks!

**Comment by AltraMayor** [comment]
This patch is ready to go upstream.

**Pull request merged**